icu4c/source/common/dictbe.cpp

```diff
@@ -265,13 +265,9 @@ ThaiBreakEngine::divideUpDictionaryRange( UText *text,
                 goto foundBest;
             }
             do {
-                int32_t wordsMatched = 1;
```

Comment:
The proposed change certainly does the same thing as the existing code, but I don't really understand this code, so I don't know what the original intent of the wordsMatched test was. Maybe we have some other bug that ought to be fixed.

Also, the same code exists in ICU4J, and whatever we do here needs to be done there as well.
https://github.com/unicode-org/icu/blob/master/icu4j/main/classes/core/src/com/ibm/icu/text/ThaiBreakEngine.java#L137